src/VisualStudio/Core/Def/Implementation/MoveToNamespace/VisualStudioMoveToNamespaceOptionsService.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.MoveToNamespace;
+using Microsoft.CodeAnalysis.Notification;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.VisualStudio.Language.Intellisense;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.MoveToNamespace
+{
+    [Export(typeof(IMoveToNamespaceOptionsService)), Shared]
+    internal class VisualStudioMoveToNamespaceOptionsService : IMoveToNamespaceOptionsService
+    {
+        private readonly IGlyphService _glyphService;
+        private readonly IThreadingContext _threadingContext;
+
+        [ImportingConstructor]
+        public VisualStudioMoveToNamespaceOptionsService(IGlyphService glyphService, IThreadingContext threadingContext)
+        {
+            _glyphService = glyphService;
+            _threadingContext = threadingContext;
+        }
+
+        public async Task<MoveToNamespaceOptionsResult> GetChangeNamespaceOptionsAsync(
+            ISyntaxFactsService syntaxFactsService,
+            INotificationService notificationService,
+            string defaultNamespace,
+            CancellationToken cancellationToken)
+        {
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
❗️ Missing `cancellationToken` use:

```suggestion
            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
            cancellationToken.ThrowIfCancellationRequested();
```